src/main/java/com/uber/cadence/internal/worker/Poller.java

```diff
@@ -272,7 +272,12 @@ public void run() throws Exception {
         }
         taskExecutor.process(task);
       } finally {
-        pollSemaphore.release();
+        while (true) {
+          if (taskExecutor.hasCapacity()) {
+            pollSemaphore.release();
+            break;
+          }
```

Comment:
can we add an else branch here to sleep for 100ms? Otherwise this will become a busy loop for a long time. 

Also need to wire up  pollerOptions.pollOnlyIfHasCapacity

Reply:
fixed